src/main/kotlin/eu/europa/ec/eudi/sdjwt/vc/SdJwtVcVerifier.kt

```diff
@@ -140,13 +140,42 @@ class SdJwtVcVerifier(
         unverifiedSdJwt: JsonObject,
         challenge: JsonObject? = null,
     ): Result<Pair<SdJwt.Presentation<JwtAndClaims>, JwtAndClaims?>> = coroutineScope {
-        val jwtSignatureVerifier = async { jwtSignatureVerifier() }
+        val jwtSignatureVerifier = async { sdJwtVcSignatureVerifier(httpClientFactory, trust, lookup) }
         val keyBindingVerifier = KeyBindingVerifier.forSdJwtVc(challenge)
         SdJwtVerifier.verifyPresentation(jwtSignatureVerifier.await(), keyBindingVerifier, unverifiedSdJwt)
     }
 
-    private fun jwtSignatureVerifier(): JwtSignatureVerifier =
-        sdJwtVcSignatureVerifier(httpClientFactory, trust, lookup)
+    companion object {
+
+        /**
+         * Creates a new [SdJwtVcVerifier] with SD-JWT-VC Issuer Metadata resolution enabled, and optionally DID resolution enabled.
+         */
+        operator fun invoke(
```

Comment:
Given that Kotlin naturally supports default values to arguments, IMHO we should

- Either, have a single method `invoke`, accepting three optional arguments, all defaulting to `null`, or 
- Use (re-)name each `invoke` method. For instance, `usingX5cOrMetadata`, `usingX5c`, `usingMetadata`, `usingDID`  

In the later case, we can also mark the constructor as public (using `null` as default value for every attribute) and checking in the `init` that at least one is present.